xknx/io/tunnel.py

```diff
@@ -202,7 +203,7 @@ async def _connect_request(self) -> bool:
                 else connect.data_endpoint.addr_tuple
             )
             # Use the individual address provided by the tunnelling server
-            self._src_address = IndividualAddress(connect.identifier)
+            self._src_address = connect.crd.individual_address or IndividualAddress(0)
```

Comment:
Not sure about the `or IndividualAddress(0)` but `connect.crd.individual_address` can be `None`.

Reply:
Ah, in theory it can, in practice it can't. 
When we receive a non-tunneling CRD here it should probably raise some error. And a tunneling-CRD always has an identifier (defaulted to 0 before). 

Not sure how to implement this the best way. Maybe we can do a function/property in CRD that returns the individual address or raises when there is none.

Reply:
Change it like this?
```suggestion
            if connect.crd.individual_address is None:
                raise CommunicationError("No individual address.")
            self._src_address = connect.crd.individual_address
```

Reply:
Right... we should then also close it 🫤
I guess that's a lot of code for a non-issue...

Reply:
Maybe just revert to defaulting to 0. If this ever yields problems it can be worked out then.

Reply:
Right, I didn't change it yet so it's fine. The behavior is also the same as before the PR.